manageiq-operator/pkg/apis/manageiq/v1alpha1/manageiq_types.go

```diff
@@ -7,62 +7,131 @@ import (
 // ManageIQSpec defines the desired state of ManageIQ
 type ManageIQSpec struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	// Application name used for deployed objects
+
+	// Application name used for deployed objects (default: manageiq)
+	// +optional
 	AppName string `json:"appName"`
 
-	// admin user initial password
+	// Initial password for "admin" user (default: smartvm)
+	// +optional
 	ApplicationAdminPassword string `json:"applicationAdminPassword"`
 
-	// Domain name for the external route
-	// Used for external authentication configuration
+	// Domain name for the external route. Used for external authentication configuration
 	ApplicationDomain string `json:"applicationDomain"`
 
-	// Application region number
+	// Database region number (default: 0)
+	// +optional
 	DatabaseRegion string `json:"databaseRegion"`
-	// Containerized database volume size
+
+	// Database volume size (default: 15Gi)
+	// +optional
 	DatabaseVolumeCapacity string `json:"databaseVolumeCapacity"`
+
+	// Secret containing the database access information, content generated if not provided (default: postgresql-secrets)
```

Comment:
My only concern is for the DatabaseSecret, KafakaSecret and TLSSecert, does it make sense to mention this in document that it is better to customize this in production environment?

Another question but not yours, I found that for those secrets, we only set a secret name but I did not see any logic for how the secret was generated, like the logic using username and password to generate etc, am I missing anything?

Reply:
> My only concern is for the DatabaseSecret, KafakaSecret and TLSSecert, does it make sense to mention this in document that it is better to customize this in production environment?

I think the only one that really *needs* to be customized in prod is the TLS one as the default would be self-signed. The others really only need to be changed if you're using a database or kafka outside of the project. Actually this reminded me that I need to open an issue to validate the secrets and to not deploy the database/kafka if an external hostname is provided in the secret.

> Another question but not yours, I found that for those secrets, we only set a secret name but I did not see any logic for how the secret was generated, like the logic using username and password to generate etc, am I missing anything?

The password and encryption key is generated [here](https://github.com/ManageIQ/manageiq-pods/blob/fe4c1445967f0abc47559baf920666aabedced74/manageiq-operator/pkg/helpers/miq-components/generators.go) and is called [here for PG](https://github.com/ManageIQ/manageiq-pods/blob/fe4c1445967f0abc47559baf920666aabedced74/manageiq-operator/pkg/helpers/miq-components/postgresql.go#L19). I think exposing the format of the secret would be good. Not sure where the best place to do that would be though. I don't think there's any reason to allow the user to specify a way to generate the password (if that's what you're asking).

Reply:
Good to know, thanks @carbonin 

> I think the only one that really needs to be customized in prod is the TLS one as the default would be self-signed. 

Do you want to fix this in a follow up?

> I don't think there's any reason to allow the user to specify a way to generate the password (if that's what you're asking).

Yes, I did not mean specify a way to generate the password, but just mean if you're using a database or kafka outside of the project, I think you have answered my question.

Reply:
> Do you want to fix this in a follow up?

What kind of fix are you thinking about? Did you want to try to implement a way of generating a signed cert? I figured someone would bring their own cert and key files then just run `oc create secret tls tls-secret --cert=tls.crt --key=tls.key`

Reply:
Opened an issue to deal with documenting secret formats and validating user-provided secret content https://github.com/ManageIQ/manageiq-pods/issues/452